### ISSUE TITLE
Crash report about the instruction `table.init`

### ISSUE THREAD

**Issue by erxiaozhou** [WasmEdge/WasmEdge#2744]
### Current State

```
Segmentation fault (core dumped)
```

### Expected

Raise an exception indicating "OOB"

## Environment

- Hardware Architecture: amd64 
- Operating system: Ubuntu 20.04
I compile the code with commit id e4196d34b1b6b2e70e4a5bd9fe84e6936e11fc85
compile:
```shell
export CC=/usr/bin/clang-14
export CXX=/usr/bin/clang++-14
rm -rf build ; mkdir build ; 
cd build ; cmake -DCMAKE_BUILD_TYPE=Debug -DWASMEDGE_BUILD_AOT_RUNTIME=OFF .. ; make -j40 ; cd ..
```
## Steps to Reproduce
[table.init_WasmEdge_crash.zip](https://github.com/WasmEdge/WasmEdge/files/12366335/table.init_WasmEdge_crash.zip)

command:
```shell
wasmedge/wasmedge --reactor <test case> to_test

```

**Comment by hydai** [comment]
An internal handler should catch this and not trigger the segmentation fault.

![CleanShot 2023-08-17 at 15 19 15](https://github.com/WasmEdge/WasmEdge/assets/2776756/7b66e16b-dcb1-4f35-8328-44900bddc3df)

**Comment by hydai** [comment]
Hi @Mash707 
You will need to build wasmedge with debug mode, use `gdb` to locate the segmentation fault, and modify the codes to make it work.

**Comment by Mash707** [comment]
> Hi @Mash707 
> 
> You will need to build wasmedge with debug mode, use `gdb` to locate the segmentation fault, and modify the codes to make it work.

Could you please tell how to build wasmedge in debug mode or refer to some docs. Thank you

**Comment by hydai** [comment]
https://wasmedge.org/docs/contribute/source/os/linux#build-wasmedge

Change the `release` to `debug` in the cmake option.

**Comment by Mash707** [comment]
Hii @hydai I built wasmedge using `debug` mode . I ran the `table.init_WasmEdge_crash.zip` using gdb and got the following output.
![1](https://github.com/WasmEdge/WasmEdge/assets/114072061/4bb92ced-b60a-4e4c-94b3-4d375ed089ad)
Then I set a breakpoint at line `1828`. Using the `next` command I started traversing line by line and found where the issue is arising. Here is a snapshot of that.
![2](https://github.com/WasmEdge/WasmEdge/assets/114072061/624c5111-2029-45fd-81c1-5d861d020677)
I am also attaching a text file for the output logs related to all the steps that I have executed so far.
[logs.txt](https://github.com/WasmEdge/WasmEdge/files/12428783/logs.txt)

**Comment by Mash707** [comment]
![image](https://github.com/WasmEdge/WasmEdge/assets/114072061/672c15eb-e274-4449-ba72-bdfd0c2f0dfa)
I also used the `backtrace` to see the call stack.

**Comment by hydai** [comment]
Fixed by #2756

**Comment by erxiaozhou** [comment]
> Hii @erxiaozhou I am still getting the `segmentation fault (core dumped)` in the error message after the PR was merged. ![image](https://user-images.githubusercontent.com/114072061/263292818-fcc12c51-b281-403a-bb23-677609605079.png) Could please share your snapshots. Just wanna confirm.

![image](https://github.com/WasmEdge/WasmEdge/assets/32102519/bb3581b1-9db9-42c1-b090-b4de69e41dde)
Hi, you can pull the latest version of the code, re-build wasmedge, and try again.
